### PR TITLE
ExportPathExcludesCommand: Allow exporting to not yet existing files …

### PR DESCRIPTION
**Pull request by fviernau**
…/ dirs

Signed-off-by: Frank Viernau <frank.viernau@here.com>

### CODE REVIEW

File: helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt [hunk 1]
```diff
@@ -75,7 +77,10 @@ internal class ExportPathExcludesCommand : CommandWithHelp() {
             .replaceConfig(repositoryConfigurationFile.readValue())
             .getRepositoryPathExcludes()
 
-        val globalPathExcludes = pathExcludesFile.readValue<RepositoryPathExcludes>()
+        @Suppress("TooGenericExceptionCaught")
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt]: Isn't this another topic than the one mentioned in the commit message? This is not about writing to non-existing *output* files / dirs, but about not requiring this *input* file. So I'd prefer to either generalize the commit message, or split this out into a new commit.

In any case, a more elegant solution would be to explicitly test for the file existence:

        val globalPathExcludes = if (pathExcludesFile.isFile) {
            pathExcludesFile.readValue<RepositoryPathExcludes>()
        } else {
            mapOf()
        }

That way you also don't have to suppress the *detekt* finding.

Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt]: > Isn't this another topic than the one mentioned in the commit message? This is not about writing to non-existing _output_ files / dirs, but about not requiring this _input_ file.

This is not an _input_ file but an _output_ file.

File: helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt [hunk 2]
```diff
@@ -75,7 +75,11 @@ internal class ExportPathExcludesCommand : CommandWithHelp() {
             .replaceConfig(repositoryConfigurationFile.readValue())
             .getRepositoryPathExcludes()
 
-        val globalPathExcludes = pathExcludesFile.readValue<RepositoryPathExcludes>()
+        val globalPathExcludes = if (pathExcludesFile.isFile) {
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt]: "target file" in the commit message sounds like a file to write to. Better say "Do not require a global path excludes file to exist".

File: helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt (hunk 1 above)
Comment (sschuberth) [reply helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt]: > This is not an _input_ file but an _output_ file,

I'm confused. You do `readValue<RepositoryPathExcludes>()`, so you *read* from that file. How can it then be an output file?

File: helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt (hunk 2 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt]: Well, the `global path excludes file` actually is being written to and for that reason I called it `target file`. Done anyway - to me this change request feels quite unimportant, do you feel the same? ..and if yes can we make such change requests optional in the future and mark them as such to speed-up the review process a bit ?

Comment (sschuberth) [reply helper-cli/src/main/kotlin/commands/ExportPathExcludesCommand.kt]: Actually, to me it just sounded wrong, and I though you might have confused the message with the other commit about the "target dir" and thus were accidentally using the word "target" here, too. So, thanks for fixing this up.

**Comment by fviernau** [comment]
Only the appveyor checks fail which is unrelated, so merge regardless of them.

**Pull request merged**